pkg/serverless/daemon/routes.go

```diff
@@ -53,6 +55,9 @@ type StartInvocation struct {
 
 func (s *StartInvocation) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Debug("Hit on the serverless.StartInvocation route.")
+	s.daemon.ExecutionSpanStateLock.Lock()
+	defer s.daemon.ExecutionSpanStateLock.Unlock()
+	s.daemon.hitOnStart = true
```

Comment:
Since this function is rather slow, it would be nice not to lock for the entire time. Either remove the defer from the unlock and do it manually after the variable assignment, or you could do something fancy like:

```go
defer func() {
	s.daemon.ExecutionSpanStateLock.Lock()
	defer s.daemon.ExecutionSpanStateLock.Unlock()
	s.daemon.hitOnStart = true
}()
```